libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/internal/navigation/HttpClient.kt

```diff
@@ -0,0 +1,76 @@
+package com.mapbox.services.android.navigation.v5.internal.navigation
+
+import com.mapbox.navigator.HttpCode
+import com.mapbox.navigator.HttpInterface
+import com.mapbox.navigator.HttpResponse
+import com.mapbox.services.android.navigation.BuildConfig
+import java.io.ByteArrayOutputStream
+import okhttp3.Interceptor
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import okhttp3.logging.HttpLoggingInterceptor
+import okio.Okio
+import timber.log.Timber
+
+internal class HttpClient(
+    private val userAgent: String,
+    private val acceptGzipEncoding: Boolean = false
+) : HttpInterface() {
+
+    private var client: OkHttpClient? = null
+    private val clientBuilder = OkHttpClient.Builder()
+
+    init {
+        check(userAgent.isNotEmpty()) {
+            "Empty UserAgent is not allowed"
+        }
+
+        if (BuildConfig.DEBUG) {
+            val interceptor = HttpLoggingInterceptor(HttpLoggingInterceptor.Logger { message ->
+                Timber.d(message)
+            }).setLevel(HttpLoggingInterceptor.Level.BASIC)
+
+            clientBuilder.addInterceptor(interceptor)
+        }
+    }
+
+    override fun isGzipped(): Boolean {
+        return acceptGzipEncoding
+    }
+
+    override fun get(url: String): HttpResponse {
+        if (client == null) {
+            client = clientBuilder.build()
+        }
+
+        val requestBuilder = Request.Builder()
+            .addHeader("User-Agent", userAgent)
+            .url(url)
+
+        if (acceptGzipEncoding) {
+            requestBuilder.addHeader("Accept-Encoding", "gzip")
+        }
+
+        client!!.newCall(requestBuilder.build()).execute().use { response ->
+            val outputStream = ByteArrayOutputStream()
+            val result = if (response.isSuccessful) HttpCode.SUCCESS else HttpCode.FAILURE
+
+            response.body()?.let { body ->
+                val sink = Okio.buffer(Okio.sink(outputStream))
+                sink.writeAll(body.source())
+                sink.close()
+            }
+
+            // FIXME core should receive Array, not List. It is List now because of bindgen
+            val bytes = outputStream.toByteArray().toList()
+            outputStream.close()
+
+            return HttpResponse(bytes, result)
+        }
+    }
+
+    // for tests only
+    fun addInterceptor(interceptor: Interceptor) {
```

Comment:
If the `OkHttpClient.Builder` is passed in the constructor, we could avoid having to expose a method used solely for testing.

Reply:
fixed